### PR TITLE
tart: init at 0.32.1

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/cirruslabs/tart

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by techknowlogick** [comment]
0.5.0 has been released, and so this PR has been changed to init at that version

**Comment by techknowlogick** [comment]
tart 0.6.0 has been released, and so this PR has been changed to init at that version

**Comment by techknowlogick** [comment]
tart 0.7.0 has been released, and so this PR has been changed to init at that version

**Comment by techknowlogick** [comment]
tart 0.12.0 is out, this PR been updated to that version

**Comment by techknowlogick** [comment]
tart 0.14.0 is out, this PR been updated to that version

**Comment by ck3d** [comment]
Can you please update to latest version.

**Comment by techknowlogick** [comment]
@ck3d updated :)

**Comment by ck3d** [comment]
Thanks for the update. During my first look to your PR I missed to recognize that you install a pre-compiled binary. We should always build from source but I have no experience with Swift in Nixpkgs.

File: pkgs/tools/virtualization/tart/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "tart";
+  version = "0.19.0";
+
+  src = fetchurl {
```
Comment (smancill) [inline pkgs/tools/virtualization/tart/default.nix:7]: You can use `fetchzip` (with `stripRoot = false`) and remove the `unpackPhase`.

File: pkgs/tools/virtualization/tart/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  unpackPhase = ''
+    mkdir source/
+    tar xvf $src -C source/
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -Dm755 source/tart -t $out/bin
```
Comment (smancill) [inline pkgs/tools/virtualization/tart/default.nix:23]:
```suggestion
    install -Dm755 tart -t $out/bin
```

`install -D` will create the target directory.

File: pkgs/tools/virtualization/tart/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
```
Comment (smancill) [inline pkgs/tools/virtualization/tart/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

No need to use a compiler if nothing is built.

File: pkgs/tools/virtualization/tart/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "macOS VMs on Apple Silicon to use in CI and other automations";
+    homepage = "https://github.com/cirruslabs/tart";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ techknowlogick ];
+    platforms = [ "aarch64-darwin"];
```
Comment (smancill) [inline pkgs/tools/virtualization/tart/default.nix:33]:
```suggestion
    platforms = [ "aarch64-darwin"];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

File: pkgs/tools/virtualization/tart/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "sha256-q7zihIaVyN42xif67PGDuSB+vq1dlM2wGVn+oaa1itE=";
+  };
+
+  dontBuild = true;
+
+  unpackPhase = ''
+    mkdir source/
+    tar xvf $src -C source/
+  '';
+
```
Comment (smancill) [inline pkgs/tools/virtualization/tart/default.nix:18]:
```suggestion
  dontConfigure = true;
  dontBuild = true;
```

**Review by smancill — CHANGES_REQUESTED** [review]

**Comment by smancill** [comment]
Rebase on top of `master`, and there is a new version available to download.

**Comment by techknowlogick** [comment]
@smancill thanks for that feedback. I've updated :)

File: pkgs/tools/virtualization/tart/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "macOS VMs on Apple Silicon to use in CI and other automations";
+    homepage = "https://github.com/cirruslabs/tart";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ techknowlogick ];
+    platforms = [ "aarch64-darwin"];
```
Comment (smancill) [inline pkgs/tools/virtualization/tart/default.nix:29]:
```suggestion
    platforms = [ "aarch64-darwin" ];
```

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -10719,6 +10719,8 @@ with pkgs;
 
   tarssh = callPackage ../servers/tarssh { };
 
+  tart = callPackage ../tools/virtualization/tart {};
```
Comment (smancill) [inline pkgs/top-level/all-packages.nix:10722]:
```suggestion
  tart = callPackage ../tools/virtualization/tart { };
```

**Review by smancill — CHANGES_REQUESTED** [review]
Sorry, I missed these small formatting changes before.

**Comment by techknowlogick** [comment]
@smancill thanks :) I've rebased, and resolved the whitespace issues

**Review by smancill — APPROVED** [review]
The derivation looks OK to me.

I am not really sure if a pre-compiled binary like this should go into nixpkgs (for a single architecture), but I'll approve anyway, so others can look into it.

**Comment by techknowlogick** [comment]
`0.32.1` has been released, this PR has been bumped to that version.

**Comment by Murderlon** [comment]
What was the reason this PR was closed? I'm interested in tart as a nix pkg. Willing to contribute a PR with these changes as a starting point if someone can confirm.

**Comment by techknowlogick** [comment]
@Murderlon Tbh it is a boring reason for closing. The need for my use of tart ended (more so paused for an undetermined amount of time). Happy if you'd like to take it over. I think one important change (other than a more recent version) is the updated license.

**Comment by emilytrau** [comment]
I've revived this PR for v1.5.0 at #237895

**Pull request closed without merging**